src/test/java/TypewiseAlert/TypewiseAlertTest.java

```diff
@@ -4,12 +4,31 @@
 
 import org.junit.Test;
 
-public class TypewiseAlertTest 
-{
-    @Test
-    public void infersBreachAsPerLimits()
-    {
-      assertTrue(TypewiseAlert.inferBreach(12, 20, 30) ==
-        TypewiseAlert.BreachType.TOO_LOW);
-    }
+
+public class TypewiseAlertTest {
+	@Test
+	public void infersBreachAsPerLimits() {
+		assertTrue(TypewiseAlert.inferBreach(12, 20, 30) == BreachType.TOO_LOW);
+		assertTrue(TypewiseAlert.inferBreach(40, 30, 40) == BreachType.NORMAL);
+		assertTrue(TypewiseAlert.inferBreach(55, 45, 50) == BreachType.TOO_HIGH);
+	}
+	
+	@Test
+	public void verifyBreachType() {
+		assertTrue(TypewiseAlert.classifyTemperatureBreach(CoolingType.PASSIVE_COOLING, -10) == BreachType.TOO_LOW);
+		assertTrue(TypewiseAlert.classifyTemperatureBreach(CoolingType.MED_ACTIVE_COOLING, 15) == BreachType.NORMAL);
+		assertTrue(TypewiseAlert.classifyTemperatureBreach(CoolingType.HI_ACTIVE_COOLING, 55) == BreachType.TOO_HIGH);
+	}
+	 @Test
+	 public void verifyEmailAlertSent() {
+		 BatteryCharacter batteryChar = new BatteryCharacter(CoolingType.HI_ACTIVE_COOLING, "bosch");
+		 assertTrue(TypewiseAlert.checkAndAlert(AlertTarget.TO_EMAIL, batteryChar, 55));
```

Comment:
how do we verify if the intended message has been received by the right target.
Test should be present for that too.